android/java/org/chromium/chrome/browser/toolbar/top/BraveToolbarLayoutImpl.java

```diff
@@ -454,6 +478,17 @@ protected void onNativeLibraryReady() {
             mBraveRewardsNativeWorker.addPublisherObserver(this);
             mBraveRewardsNativeWorker.getAllNotifications();
         }
+
+        if (ChromeFeatureList.isEnabled(
+                BraveFeatureList.BRAVE_PICTURE_IN_PICTURE_FOR_YOUTUBE_VIDEOS)) {
+            try {
+                final FullscreenManager fullscreenManager =
+                        BraveActivity.getBraveActivity().getFullscreenManager();
```

Comment:
We are going to get rid of `getBraveActivity()` function in the future as it could be null and destroyed in some cases. I would think that a more correct way is to get it from `mToolbarDataProvider` of a super class after `initialize` method is called. It has an access to a tab and we can get a corresponding Activity from there `mToolbarDataProvider.getTab().getWindowAndroid().getActivity()`. After that we can check `if (activity instanceof ChromeActivity)` and get the `FullscreenManager` from it.

Reply:
actually `getToolbarDataProvider` will work :)

Reply:
yep, that's even better!